### PR TITLE
Replaced offline service by worker and removed static list.

### PR DESCRIPTION
**Pull request by Lavanyagaur22**
### Description

I have replaced OfflineSyncService by `OfflineSyncWorker` as there were some Android O and above related problems in case of foreground service when we are in the background. 
With worker, it's running perfectly fine for all test cases.

Also, I have removed the static array list of callbacks and mutations which we were using earlier to make a call to the server when in the foreground. Instead, each time we have fetched the item from DB and when each mutation is processed we have deleted it simultaneously from the database.

### CODE REVIEW

**Comment by Lavanyagaur22** [comment]
@harshithdwivedi Please have a look at it.

File: offix/src/main/java/org/aerogear/offix/OfflineList.kt [hunk 1]
```diff
@@ -4,19 +4,11 @@ import com.apollographql.apollo.ApolloCall
 import com.apollographql.apollo.ApolloClient
 import com.apollographql.apollo.api.Operation
 
-/* This class helps in creating an array list of mutations and callbacks associated with the apollo call.
+/* This class helps in getting access to the instance of apollo client provided by the user.
+   Also, in future the static objects can be placed here.
  */
 class OfflineList {
 
-    /* Array List to store mutation objects.
```
Comment (harshithdwivedi) [inline offix/src/main/java/org/aerogear/offix/OfflineList.kt:11]: Can we remove/rename this class?

File: offix/src/main/java/org/aerogear/offix/OfflineSyncWorker.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            override fun onResponse(response: Response<Operation.Data>) {
+                Log.d(TAG, response.data().toString())
+            }
+        }
+
+        Log.d(TAG, " Size of database list ** : ${mutationDao?.getAllMutations()?.size}")
+
+        mutationList?.forEach { storedmutation ->
+
+            if (Offline.isNetwork()) {
```
Comment (harshithdwivedi) [inline offix/src/main/java/org/aerogear/offix/OfflineSyncWorker.kt:45]: Have an else block here where you return Result.retry()

**Review by harshithdwivedi — CHANGES_REQUESTED** [review]
Looks good, minor changes needed

**Comment by Lavanyagaur22** [comment]
@harshithdwivedi Done👍

File: offix/src/main/java/org/aerogear/offix/OfflineSyncWorker.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+                apollcall?.enqueue(apolloCallback)
+
+                /*
+                 Remove processed mutation.
+                */
+                mutationDao?.deleteMutation(storedmutation)
+            }
+        }
+        return if (mutationDao?.getAllMutations()?.isEmpty()!!) {
```
Comment (harshithdwivedi) [inline offix/src/main/java/org/aerogear/offix/OfflineSyncWorker.kt:63]: This is good

**Review by harshithdwivedi — APPROVED** [review]

**Pull request merged**